internal/verification/authenticate_stats.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verification
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/google/exposure-notifications-server/internal/verification/database"
+	"github.com/google/exposure-notifications-server/internal/verification/model"
+	"github.com/google/exposure-notifications-server/pkg/logging"
+)
+
+// AuthenticateStatsToken parse the provided JWT and determines if it is an authorized stats request.
+func (v *Verifier) AuthenticateStatsToken(ctx context.Context, rawToken string) (int64, error) {
+	logger := logging.FromContext(ctx)
+
+	var healthAuthorityID int64
+	var claims *jwt.StandardClaims
+
+	token, err := jwt.ParseWithClaims(rawToken, &jwt.StandardClaims{}, func(token *jwt.Token) (interface{}, error) {
```

Comment:
I think @whaught found an edge case using `ParseWithClaims` in the verification server, Do you remember what it was @whaught?

/hold

Reply:
I do! ParseWithClaims automatically check claims.Valid(), but if you wish to validate yourself you can set Skip on the parser. That said I don't think we are manually checking claim validity here

Reply:
See: https://github.com/google/exposure-notifications-verification-server/pull/1465/files

Reply:
ok, there's no issue here then.